WalletWasabi.Backend/Startup.cs

```diff
@@ -69,10 +69,16 @@ public void ConfigureServices(IServiceCollection services)
 		services.AddLogging(logging => logging.AddFilter((s, level) => level >= Microsoft.Extensions.Logging.LogLevel.Warning));
 
 		services.AddSingleton<IExchangeRateProvider>(new ExchangeRateProvider());
+		services.AddSingleton(serviceProvider =>
+		{
+			string configFilePath = Path.Combine(dataDir, "Config.json");
+			Config config = new(configFilePath);
+			config.LoadOrCreateDefaultFile();
+			return config;
+		});
+
 		services.AddSingleton<IdempotencyRequestCache>();
-#pragma warning disable CA2000 // Dispose objects before losing scope, reason: https://github.com/dotnet/roslyn-analyzers/issues/3836
-		services.AddSingleton(new Global(Configuration["datadir"]));
-#pragma warning restore CA2000 // Dispose objects before losing scope
+		services.AddSingleton(serviceProvider => ActivatorUtilities.CreateInstance<Global>(serviceProvider, dataDir));
```

Comment:
I would prefer more explicit creation, i think is better.

```c#
		services.AddSingleton(serviceProvider => {
		var rpc =  provider.GetRequiredService<IRpcClient>();
		var cfg =  provider.GetRequiredService<Config>();
		.....
		return new Global(datadir, rpc, cfg):
```

Reply:
I don't have a strong opinion, I would not involve ActivatorUtilities in this - reason: readability for rookies.